src/main/java/ch/njol/skript/expressions/ExprSpawnEggEntity.java

```diff
@@ -0,0 +1,123 @@
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntitySnapshot;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.SpawnEggMeta;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Spawn Egg Entity")
+@Description({
+	"Gets or sets the entity snapshot that the provided spawn eggs will spawn when used.",
+	"NOTE: When using an entity data i.e 'a zombie', it will spawn a new zombie to retrieve an entity snapshot and then be removed."
+})
+@Examples({
+	"set {_item} to a zombie spawn egg",
+	"broadcast the spawn egg entity of {_item}",
+	"",
+	"spawn a pig at location(0,0,0):",
+		"\tset the max health of entity to 20",
+		"\tset the health of entity to 20",
+		"\tset {_snapshot} to the entity snapshot of entity",
+		"\tclear entity",
+	"set the spawn egg entity of {_item} to {_snapshot}",
+	"if the spawn egg entity of {_item} is {_snapshot}: # Minecraft 1.20.5+",
+	"",
+	"set the spawn egg entity of {_item} to (random element out of all entities)",
+	"",
+	"set the spawn egg entity of {_item} to a zombie"
+})
+@RequiredPlugins("Minecraft 1.20.2+, Minecraft 1.20.5+ (comparisons)")
+@Since("INSERT VERSION")
+public class ExprSpawnEggEntity extends SimplePropertyExpression<Object, Object> {
+
+	static {
+		if (Skript.classExists("org.bukkit.entity.EntitySnapshot"))
+			register(ExprSpawnEggEntity.class, Object.class, "spawn egg entity", "itemstacks/itemtypes/slots");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (Player.class.isAssignableFrom(exprs[0].getReturnType())) {
+			Skript.error("You can't set the spawn egg entity to a player.");
+			return false;
+		}
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	public @Nullable EntitySnapshot convert(Object object) {
+		ItemStack itemStack = ItemUtils.asItemStack(object);
+		if (itemStack == null || !(itemStack.getItemMeta() instanceof SpawnEggMeta eggMeta))
+			return null;
+		return eggMeta.getSpawnedEntity();
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET)
+			return CollectionUtils.array(EntitySnapshot.class, Entity.class, EntityData.class);
+		return null;
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		if (delta == null || delta[0] == null)
+			return;
+		EntitySnapshot snapshot = null;
+		if (delta[0] instanceof EntitySnapshot entitySnapshot) {
+			snapshot = entitySnapshot;
+		} else if (delta[0] instanceof Entity entity) {
+			snapshot = entity.createSnapshot();
+		} else if (delta[0] instanceof EntityData<?> entityData) {
+			Location location = new Location(Bukkit.getWorlds().get(0), 0, 0, 0);
+			Entity entity = entityData.spawn(location);
+			snapshot = entity.createSnapshot();
+			entity.remove();
```

Comment:
Is this the best way to do this? Would EntityFactory#createEntitySnapshot(String) be useful?